plugins/handlers/gallery.py

```diff
@@ -285,17 +359,26 @@ def safe_resize(source, destination, size):
 
     This makes use of `RESIZE_METHODS` to determine which resize handler to use.
     """
-    resize_method = evaluate_tests(RESIZE_METHODS, os.path.basename(source))
+    resize_method = evaluate_tests(RESIZE_METHODS, {"basename": os.path.basename(source)})
     resize_method(source,
                   destination,
                   f"{size.width}x{size.height}")
 
 
+# TODO: Remove this.
+def resize_simple(source, destination, size):
+    destination_root, destination_basename = os.path.split(destination)
+    return resize(source, destination_root, "", destination_basename, size, 1)
+
+
+# TODO: Write a utility to convert a mimetype to a path with an extension (and test it).
```

Comment:
Perhaps this functionality could be used external to the transform operation, rather than internally?